front_loader.py

```diff
@@ -0,0 +1,112 @@
+import requests
+from models import *
+from keys.api_key import api_key
+from app import app
+
+connect_db(app)
+
+db.drop_all()
+db.create_all()
+
+
+
+def do_ingredients_exist(drink):
+    for i in range(1,16):
+        ingredient = drink[f'strIngredient{i}']
+        if not ingredient:
+            return True
+        exists = Ingredient.query.filter(Ingredient.name == ingredient).first()
+        if not exists:
+            return False
+    return True
+
+def create_ingredients(drink):
+    for i in range(1,16):
+        ingredient = drink[f'strIngredient{i}']
+        if ingredient:
+            exists = Ingredient.query.filter(Ingredient.name == ingredient).first()
+            if not exists:
+                new_ingredient = Ingredient(name=ingredient)
+                db.session.add(new_ingredient)
+                db.session.commit()
+
+def fix_instrucions(instr):
+    if instr is not None:
+        return instr.replace('\\r', '')
+    else:
+        return 'Not yet provided'
+
+def fix_url(url):
+    return url.replace('\\', '')
+
+def is_alcoholic(str):
+    return str.lower() == 'alcoholic'
+
+def create_recipe(drink):
+
+    new_recipe = Recipe(
+                        id=drink['idDrink'],
+                        name=drink['strDrink'],
+                        category=drink['strCategory'],
+                        process_EN=fix_instrucions(drink['strInstructions']),
+                        process_ES=fix_instrucions(drink['strInstructionsES']),
+                        process_FR=fix_instrucions(drink['strInstructionsFR']),
+                        process_DE=fix_instrucions(drink['strInstructionsDE']),
+                        process_IT=fix_instrucions(drink['strInstructionsIT']),
+                        image_url=fix_url(drink['strDrinkThumb']),
+                        alcoholic=is_alcoholic(drink['strAlcoholic']),
+                        created_on=drink['dateModified']
+    )
+
+    try:
+        db.session.add(new_recipe)
+        db.session.commit()
+        print("drink added")
+    except Exception as err:
+        print('error', err)
+
+
+def add_recipe_ingredients_relationships(drink):
+    for i in range(1,16):
```

Comment:
"Magic number" 16 - please avoid magic numbers like this. They are confusing and make it difficult to reason about your code.

Reply:
I think maybe I should just leave a comment? The reasoning here is that the API leaves room for 15 ingredients, and 15 corresponding measurements.  It rarely uses more than 10.

Or I guess I could just change the loop to be a while, and set it to break when it encounters a None